Tests/UnitTests/EdgeHitProcessorTests.swift

```diff
@@ -677,41 +788,64 @@ class EdgeHitProcessorTests: XCTestCase {
                                         },
                                         getLocationHint: { return nil })
 
-        mockNetworkService?.connectAsyncMockReturnConnection = HttpConnection(data: "{}".data(using: .utf8), response: HTTPURLResponse(url: url, statusCode: 200, httpVersion: nil, headerFields: nil), error: nil)
+        mockNetworkService.setMockResponse(
+            url: CONSENT_ENDPOINT,
+            httpMethod: .post,
+            responseConnection: HttpConnection(
+                data: "{}".data(using: .utf8),
+                response: HTTPURLResponse(
+                    url: url,
+                    statusCode: 200,
+                    httpVersion: nil,
+                    headerFields: nil),
+                error: nil))
 
         let edgeEntity = EdgeDataEntity(event: consentUpdateEvent, identityMap: [:])
         let entity = DataEntity(uniqueIdentifier: "test-uuid", timestamp: Date(), data: try? JSONEncoder().encode(edgeEntity))
 
         // test
-        assertProcessHit(entity: entity, sendsNetworkRequest: true, returns: true)
+        assertProcessHit(entity: entity, urlString: CONSENT_ENDPOINT, sendsNetworkRequest: true, returns: true)
 
-        guard let requestPayload = mockNetworkService?.connectAsyncCalledWithNetworkRequest?.connectPayload else {
-            XCTFail("unexpected nil request connect payload")
+        guard let networkRequest = mockNetworkService.getNetworkRequestsWith(url: CONSENT_ENDPOINT, httpMethod: .post).first else {
+            XCTFail("Unable to find valid network request.")
             return
         }
 
-        let payload = asFlattenDictionary(data: requestPayload)
+        let payload = asFlattenDictionary(data: networkRequest.connectPayload)
 
         // Implementation Details are not added to Consent events
         XCTAssertNil(payload["xdm.implementationDetails.name"])
         XCTAssertNil(payload["xdm.implementationDetails.version"])
         XCTAssertNil(payload["xdm.implementationDetails.environment"])
```

Comment:
Are these tests still needed if you are validating the entire JSON request below?